### DISCUSSION TITLE
Anybody working on real-time Advanced Voice Mode support feature?

### DISCUSSION THREAD

**Discussion by DmitriyAlergant** [open-webui/open-webui#7993] [Ideas]
There is a Call button but seemingly we don't yet support actual real-time api that ChatGPT Advanced Voice Mode is using.
The OpenAI's realtime audio model is pricey but still cool, and will eventually become cheaper.

Example front-end implementation (for reference): https://tools.simonwillison.net/openai-webrtc

I guess it is a significant feature, but should be doable

* Significant frontend changes in CallOverlay.svelte, perhaps a new component CallOverlayRealtime.svelte
* Expose new backend API for "realtime" streaming
* A new model-level feature to indicate which models even support realtime audio

I wonder if anybody started working on this or planning - or the feature still up for grabs?

**Comment by security-somanos** [comment]
I hope someone is working on this!

**Comment by spammenotinoz** [comment]
I am using another project for Advanced Voice based using the OpenAI API, while the voice works extremely well, the rest of the project is complete junk.
The Realtime API experience is excellent, but you do need to watch the cost\pricing, use system prompts to keep the responses brief.

**Reply by Pl8tinium** [reply]
damn i just saw the prices, those are hefty

still, would love to see an integration of this! 

would be especially important when FOSS models catch up those realtime capabilities

**Comment by Simon-Stone** [comment]
I am very curious about that, too. Would love to collaborate with someone if there are already efforts on the way.

**Comment by peuportier** [comment]
maybe you shall unmute by kuytai that I guess can be integrated to Open-webui : https://unmute.sh

**Reply by flefevre** [reply]
Yes it will definitely a good idea.
But at present time they do not package everything.
I will try to contact them

**Comment by arturcsanntos** [comment]
reinforcing this too! OpenAI just launched a new version of the [Realtime API](https://openai.com/index/introducing-gpt-realtime/) and i think  it would be great if OWUI supported it.

**Comment by peuportier** [comment]
Yes—Unmute (Kyutai) can be integrated with Open WebUI in practice, but not as a single “drop-in” package today.
The straightforward way is to run Kyutai STT/TTS behind OpenAI-compatible endpoints and point Open WebUI’s voice (STT/TTS) settings at those URLs. A deeper “full Unmute” realtime WebSocket-style integration would require additional packaging and some Open WebUI-side work. I do hope they have the time to do it

**Comment by alexlach** [comment]
I find myself back here today after another [OpenAI Realtime API](https://openai.com/index/advancing-voice-intelligence-with-new-models-in-the-api/) update. Lots of complexity with making this happen in OWUI... but it would be soooo cool!

**Comment by rbb-dev** [comment]
It's been running in production for a while now, with a number of users relying on it day-to-day - so, in practical terms, yes.

Docker images: https://github.com/rbb-dev/open-webui/pkgs/container/open-webui-realtime

I've tried to bring it upstream, it's offered as a contribution, and I keep it continuously rebased on the latest Open WebUI, so it stays review-ready rather than drifting. It hasn't been picked up for merge so far, which is completely fair: it's a large feature and a lot to review, and the maintainers have their own roadmap. No hard feelings.

What it needs most now is wider community testing, which is hard to come by while it lives in a fork. If you give it a try, I'd genuinely value the feedback. In the meantime, I'll keep tracking upstream, and if a merge isn't on the cards, I'll repackage it as a plugin/pipe so it's easier for everyone to run.

Either way, I'm committed to keeping it current. 🙂

**Comment by spammenotinoz** [comment]
Thank-you, it actually works extremely well.
Which provider \ model do you recommend for the budget conscious.

**Comment by Pl8tinium** [comment]
@rbb-dev  i've now also tried it and am blown away. This is exactly how i was hoping for the integration to be. 🚀 

wonderful job! Also i really really appreciate your  [Open-WebUI-OpenRouter-pipe](https://github.com/rbb-dev/Open-WebUI-OpenRouter-pipe) which can finally replace my crappy pipe that just appends the :online tab behind my openrouter models so they can use their native web search capabilities.

The only thing im still missing is web search capabilities for the configured openai realtime models, but i'll look into adding a separate websearch tool for that.

Thanks for building and maintaining these two components, they really enhance my openwebui experience and tackle the main pain points i have with openwebui!!

**Reply by Pl8tinium** [reply]
I really hope that one day the realtime api changes can be merged into main
